conans/cli/commands/search.py

```diff
@@ -0,0 +1,60 @@
+import json
+
+from conans.client.output import Color
+from conans.errors import NoRemoteAvailable
+from conans.cli.cli import Extender
+from conans.cli.command import conan_command
+
+
+def output_search_cli(info, out):
+    results = info["results"]
+    for remote_info in results:
+        source = "cache" if remote_info["remote"] is None else str(remote_info["remote"])
+        out.writeln("{}:".format(source), Color.BRIGHT_WHITE)
+        for conan_item in remote_info["items"]:
+            reference = conan_item["recipe"]["id"]
+            out.writeln(" {}".format(reference))
+
+
+def output_search_json(info, out):
+    results = info["results"]
+    myjson = json.dumps(results, indent=4)
+    out.writeln(myjson)
+
+
+def apiv2_search_recipes(query, remote_patterns=None, local_cache=False):
+    remote = None
+    if remote_patterns is not None and len(remote_patterns) > 0:
+        remote = remote_patterns[0].replace("*", "remote")
+
+    if remote and "bad" in remote:
+        raise NoRemoteAvailable("Remote '%s' not found in remotes" % remote)
+
+    search_results = {"results": [{"remote": remote,
+                                   "items": [{"recipe": {"id": "app/1.0"}},
+                                             {"recipe": {"id": "liba/1.0"}}]}]}
+
+    return search_results
+
+
+@conan_command(group="Consumer commands", formatters={"cli": output_search_cli,
+                                                      "json": output_search_json})
+def search(*args, conan_api, parser, **kwargs):
```

Comment:
**kwargs unused?

Could it be ``(conan_api, parser, *args)`` instead?

Reply:
That is to absorb `commands` and `groups` arguments here: https://github.com/conan-io/conan/blob/473080621e54d469c710c255a63a1a8edc770a3e/conans/cli/commands/help.py#L42
Maybe I could call the help command separately using and if better?